torch/__init__.py

```diff
@@ -323,6 +324,56 @@ def __repr__(self):
     def get_pyobj(self):
         return self.node
 
+class SymBool:
+    """
+    Like an bool (including magic methods), but redirects all operations on the
+    wrapped node. This is used in particular to symbolically record operations
+    in the symbolic shape workflow.
+
+    Unlike regular bools, regular boolean operators will force extra guards instead
+    of symbolically evaluate.  Use the bitwise operators instead to handle this.
+    """
+
+    def __init__(self, node):
+        from torch.fx.experimental.symbolic_shapes import SymNode
+        assert isinstance(node, SymNode)
+        # This field MUST be named node; C++ binding code assumes that this
+        # class has a field named node that stores SymNode
+        self.node = node
+
+    def __bool__(self):
+        return self.node.bool_()
+
+    # Magic methods installed by torch.fx.experimental.symbolic_shapes
+    def __and__(self, other) -> "SymBool":
+        raise AssertionError("type stub not overridden")
+
+    def __or__(self, other) -> "SymBool":
+        raise AssertionError("type stub not overridden")
+
+    # This is not a real magic method but we define it anyway, because
+    # calling ~a on things that are potentially booleans will lead to
+    # very strange results.  You can trigger this with sym_not instead.
+    def __sym_not__(self) -> "SymBool":
```

Comment:
@ezyang Wait, hold on. There isn't a magic method for `not obj` but `~obj` actually has one (called `__invert__`): [__invert__](https://docs.python.org/3/reference/datamodel.html#object.__invert__)

Reply:
No, I do NOT want to override invert, because you should never actually do `~a` in a context where a could be a bool, you will silently get nonsense.